test/extended/builds/docker-build-no-outputname.go

```diff
@@ -0,0 +1,39 @@
+package builds
+
+import (
+	"fmt"
+	"path/filepath"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("builds: Docker image buildconfig without output.to.Name specified", func() {
+	defer g.GinkgoRecover()
+	var (
+		dockerImageFixture = filepath.Join("fixtures", "test-docker-no-outputname.json")
+		oc                 = exutil.NewCLI("build-docker-no-outputname", exutil.KubeConfigPath())
+	)
+
+	g.Describe("Building from a template", func() {
+		g.It(fmt.Sprintf("should create an image from %q template without an output image reference defined", dockerImageFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+			err := oc.Run("create").Args("-f", dockerImageFixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting a build")
+			buildName, err := oc.Run("start-build").Args("test").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting the build is in Complete phase")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName, exutil.CheckBuildSuccessFunc, exutil.CheckBuildFailedFunc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("expecting build to pass without an output image reference specified")
+			out, err := oc.Run("start-build").Args("test", "--follow").Output()
```

Comment:
rather than run another build, you should be able to get the logs from the previous build and check them.

also "--follow" is notoriously finicky because it sometimes times out before the build starts, resulting in no logs, so if you are going to use it, you need to wrap it in some sort of retry logic.